client/src/pages/ShoppingCartPage.js

```diff
@@ -31,6 +35,46 @@ function ShoppingCartPage(props) {
 
   const [cart, setCart] = useState(location.state.shoppingCart);
 
+  const [client, setClient] = useState({});
+
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    let clientID;
+    async function updateClient() {
+      if (authContext.currentUser && authContext.currentUser.role === UserRoles.CLIENT)
```

Comment:
Move the retrieval of the clientID outside of the function updateClient

Then rename updateClient to fetchClientInfo()

And do:
```
useEffect(()=>{
   clientID retrieval goes here
   
   define fetchClientInfo
   
   call fetchClientInfo()
}, [...])
```